docs/coding-conventions.md

```diff
@@ -0,0 +1,301 @@
+
+### Bad example
+
+```go
+func doSomething() string {
+    return "constantValue" + computeSuffix()
+}
+
+func doSomethingElse() string {
+    return computePrefix() + "constantValue"
+}
+```
+
+### Good example
+
+```go
+const constantValue = "constantValue"
+
+func doSomething() string {
+    return constantValue + computeSuffix()
+}
+
+func doSomethingElse() string {
+    return computePrefix() + constantValue
+}
+```
+
+# Stick to a uniform import order and naming scheme for package imports
+
+- Use the following import order with one block per item:
+    - Golang standard libraries
+    - Ginkgo / Gomega imports (only in test files)
+    - Third-party libraries
+    - Local packages
+- Use the following naming schemes for imports:
+    - `virtv1` for imports of `kubevirt.io/api`
+    - `metav1` for imports of `k8s.io/apimachinery/pkg/apis/meta/v1`
+    - `k8s` prefix for `k8s.io` imports on collision with other imports
+    - TODO: Add more
+- `make format` is able to help you with this.
+
+### Bad example
+
+```go
+import (
+    "context"
+    k8sv1 "k8s.io/api/core/v1"
+    "time"
+
+    ."github.com/onsi/ginkgo/v2"
+    ."github.com/onsi/gomega"
+    metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+    v1 "kubevirt.io/api/core/v1"
+    "kubevirt.io/kubevirt/tests/clientcmd"
+
+    "kubevirt.io/kubevirt/tests/testsuite"
+)
+```
+
+### Good example
+
+```go
+import (
+    "context"
+    "time"
+
+    ."github.com/onsi/ginkgo/v2"
+    ."github.com/onsi/gomega"
+
+    k8sv1 "k8s.io/api/core/v1"
+    metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+    virtv1 "kubevirt.io/api/core/v1"
+
+    "kubevirt.io/kubevirt/tests/clientcmd"
+    "kubevirt.io/kubevirt/tests/testsuite"
+)
+```
+
+## Short coding conventions (TLDR)
+
+Here are some important but short conventions. If you want to learn more
+about conventions in detail take a look at the [detailed
+coding conventions](#detailed-coding-conventions).
+
+- [Write elegant, cohesive and easily readable code](#write-elegant-cohesive-and-easily-readable-code)
+- [Avoid nesting and complexity by using early returns](#avoid-nesting-and-complexity-by-using-early-returns)
+- [Use the same coding style throughout a file](#use-the-same-coding-style-throughout-a-file)
+- [Avoid repetition of hardcoded values](#avoid-repetition-of-hardcoded-values)
+- [Stick to a uniform import order and naming scheme for package imports](#stick-to-a-uniform-import-order-and-naming-scheme-for-package-imports)
+- Inline variable assignments when possible.
+    - For example: Inline `err` checks in if-statements.
+- Use switch-cases to avoid long if/else if/else statements.
+- Isolate code where it makes sense.
+    - Use interfaces instead of bare structs.
+- Avoid use of global variables.
+    - Use structs and receiver methods to keep state.
+- Do not make files too long.
+    - Add helpers to the places in which they are used.
+    - Long files like `tests/utils.go` are hard to maintain.
```

Comment:
Sometimes, specific files for these common functions might still be needed since they are used across several other files

maybe we can suggest avoiding `utils` wording and having these functions in files with representative names would be a good way to go
p.e. `tests/libmonitoring/scaling.go` and `tests/libmonitoring/prometheus.go`

Reply:
looks good, but wdyt about reordering such as:

```
    - Avoid adding helpers in a single place. Long files like `tests/utils.go` are hard to maintain.
    - Alternatively, add helpers to the places where they are used or group them in packages/files with representative names.
```

Reply:
Makes sense